taskcat/s3_sync.py

```diff
@@ -22,33 +22,37 @@
 class S3Sync(object):
     """Syncronizes local project files with S3 based on checksums.
 
-    Excludes hidden files, unpackaged lambda source and taskcat /ci/ files.
+    Excludes hidden files, unpackaged lambda source and other paths and
+    files matching the optional "exclude" parameter(s).
     Uses the Etag as an md5 which introduces the following limitations
         * Uses undocumented etag algorithm for multipart uploads
         * Does not work wil files uploaded in the console that use SSE encryption
         * see https://docs.aws.amazon.com/AmazonS3/latest/API/RESTCommonResponseHeaders.html for more info
     Does not support buckets with versioning enabled
     """
 
-    exclude_files = [
-        ".*",
+    hidden_files_and_folders =  "*/.*"
+    lambdas_source = "functions/source/"
+
+    exclude_defaults = [
+        hidden_files_and_folders,
+        lambdas_source,
         "*.md"
     ]
-    exclude_path_prefixes = [
-        "functions/source/",
-        "."
-    ]
```

Comment:
We leverage this class for other toolsets and depend on the current class attributes (including this one). If possible - can you not modify the S3Sync class and instead set the attributes prior to uploading?  Ex:

```
S3Sync.exclude_path_prefixes.append(...)
S3Sync(...)
```

Reply:
/cc @jaymccon

Reply:
ok done